.github/workflows/pr-build.yml

```diff
@@ -3,43 +3,77 @@ on:
   push:
     branches:
       - main
-  pull_request_target:
-    branches:
-      - main
+  pull_request:
 
 jobs:
+  extract-args-java-11:
```

Comment:
This seems like excessive complexity for a simple task. In the build case, we don't even need to generate all the individual tags anyway. In the publish version, since that's the only spot we need them, you can just inline the couple `cut` calls.

Reply:
Yes it is a bit complex, but since we wouldn't want to store build args hidden in github actions yamls rather in repo source I thought this is required

Can we inline reading data from a file and doing cut to pass as a variable to an action? I couldn't think of a way to do it and built a reusable workflow

Reply:
Could you do something like this? basically just an inline form of the same code. I also moved the build arg name from inside the file to outside.

```yaml
      - name: Read Java Version
        shell: bash
        id: versions
        run: |
          ZULU_JDK_VERSION=`cat ${{ matrix.java-major-version}}/java.version}}`
          echo ZULU_JDK_VERSION=${ZULU_JDK_VERSION} >> $GITHUB_OUTPUT
          echo JAVA_MAJOR_VERSION=$(echo $ZULU_JDK_VERSION | cut -d'.' -f1)" >> $GITHUB_OUTPUT
          echo JAVA_FULL_VERSION=$(echo $ZULU_JDK_VERSION | cut -d'-' -f1) >> $GITHUB_OUTPUT
```

Then, for build args just declare it like
```yaml
build-args: |
                      JAVA_VERSION=${{steps.versions.output.ZULU_JDK_VERSION}}
```

Reply:
I also wouldn't bother publishing the zulu version tag - it should be treated as an impl detail since we're not making any guarantees about the specific distro of java we're using.

Reply:
I'm not sure why we'd add any more args, and if we did, it's likely we'd be changing the build too. If the arg was statically defined, there'd be no point in passing it in in the first place. If it were dynamic, we'd likely be using a matrix or some other way of calculating the value for it in CI then need to pass them in like above anyway.

The java version is actually bit of a strange case as it's static, but kept outside the dockerfile so it can be consumed to tag as well. A common counter example would be something like the current commit SHA.

So in other words, the more likely future case is we'd need to calculate a build arg in the build and then pass it to the build command which would be more difficult to do if merging it with the static env file. Not positive if that makes sense, or if I'm missing your point though.

Reply:
> The java version is actually bit of a strange case as it's static, but kept outside the dockerfile so it can be consumed to tag as well.

Ah, got it. I thought it was done as a good practice whatever things are not required to be static, make them as args and pass it during build. Hence I was thinking of multiple build args as a possibility.

Makes sense to me now, updating

Reply:
I'm curious of how we will be scanning in individual repos though? Matrix seems a overkill, especially for repos like api-anomaly-detection which publishes around 4 images. Will we be taking the call that scanning only the CI arch is fine and assume that the same set of vulnerabilities exist on OS level for both x86 and arm?